/**
 * Copyright 2020 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option cc_enable_arenas = true;
package mindspore;

enum DataType {
  MS_FLOAT32 = 0;
  MS_FLOAT16 = 1;
  MS_INT8 = 2;
  MS_INT32 = 3;
  MS_UINT8 = 4;
  MS_INT16 = 6;
  MS_UINT16 = 7;
  MS_UINT32 = 8;
  MS_INT64 = 9;
  MS_UINT64 = 10;
  MS_FLOAT64 = 11;
  MS_BOOL = 12;
  MS_STRING = 13;
  MS_DUAL_SUB_INT8 = 14;
  MS_DUAL_SUB_UINT8 = 15;
  MS_COMPLEX64 = 16;
  MS_COMPLEX128 = 17;
  MS_QINT8 = 18;
  MS_QINT16 = 19;
  MS_QINT32 = 20;
  MS_QUINT8 = 21;
  MS_QUINT16 = 22;
  MS_RESOURCE = 23;
  MS_STRING_REF = 24;
  MS_DUAL = 25;
  MS_UNKNOWN = 26;
}
